crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::QByteArray;
+use cxx::{type_id, ExternType};
+use std::{
+    ffi::{c_char, CString},
+    mem::MaybeUninit,
+};
+
+#[cxx::bridge]
+mod ffi {
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qbytearray.h");
+        type QByteArray = crate::QByteArray;
+
+        include!("cxx-qt-lib/qimage.h");
+        type QImage = super::QImage;
+
+        #[rust_name = "is_null"]
+        fn isNull(self: &QImage) -> bool;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qimage_drop"]
+        fn drop(image: &mut QImage);
+
+        #[doc(hidden)]
+        #[rust_name = "qimage_init_from_data"]
+        unsafe fn qimageInitFromData(data: &QByteArray, format: *const c_char) -> QImage;
+    }
+}
+
+#[repr(C)]
+pub struct QImage {
+    // Static checks on the C++ side ensure this is true.
+    // See qcolor.cpp
+    #[cfg(qt_version_major = "5")]
+    _data: MaybeUninit<[usize; 4]>,
+    #[cfg(qt_version_major = "6")]
+    _data: MaybeUninit<[usize; 3]>,
+}
+
+// Safety:
+//
+// Static checks on the C++ side to ensure the size & alignment is the same.
+unsafe impl ExternType for QImage {
+    type Id = type_id!("QImage");
+    type Kind = cxx::kind::Trivial;
+}
+
+impl Drop for QImage {
+    fn drop(&mut self) {
+        ffi::qimage_drop(self);
+    }
+}
+
+impl QImage {
+    pub fn from_data(data: &QByteArray, format: &str) -> Option<Self> {
+        let format = CString::new(format).ok()?;
+        let image = unsafe { ffi::qimage_init_from_data(data, format.as_ptr() as *const c_char) };
```

Comment:
then here you just pass in the `format` and you don't need to convert to a `CString`